app/src/main/java/com/kickstarter/viewmodels/MessageThreadHolderViewModel.java

```diff
@@ -88,7 +99,36 @@ public ViewModel(final @NonNull Environment environment) {
       this.unreadCountTextViewText = this.messageThread
         .map(MessageThread::unreadMessagesCount)
         .map(NumberUtils::format);
-      this.unreadIndicatorViewHidden = this.messageThread.map(m -> m.unreadMessagesCount() == 0);
+
+      this.unreadIndicatorViewHidden = hasUnreadMessages.map(BooleanUtils::negate);
+
+      this.messageThread
+        .compose(takeWhen(this.messageThreadCardViewClicked))
+        .subscribe(thread -> markedAsRead(thread, this.sharedPreferences));
+    }
+
+    private static @NonNull String cacheKey(final @NonNull MessageThread messageThread) {
+      return SharedPreferenceKey.MESSAGE_THREAD_HAS_UNREAD_MESSAGES + messageThread.id();
+    }
+
+    private static boolean hasUnreadMessages(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+      return sharedPreferences.getBoolean(cacheKey(messageThread), messageThread.unreadMessagesCount() > 0);
+    }
+
+    private static void markedAsRead(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+      final SharedPreferences.Editor editor = sharedPreferences.edit();
+      editor.putBoolean(cacheKey(messageThread), false);
+      editor.apply();
+    }
+
+    private static void setHasUnreadMessagesPreference(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+
+      final SharedPreferences.Editor editor = sharedPreferences.edit();
+      editor.putBoolean(cacheKey(messageThread), messageThread.unreadMessagesCount() > 0);
+      editor.apply();
```

Comment:
just wanna confirm that we don't want to add `MESSAGE_THREAD_HAS_UNREAD_MESSAGES` to the initialization of other `SharedPreferences` in the ApplicationModule? Seems like we're just using the `sharedPreferences` here as key/value store for these threads, but wanted to confirm that's...ok.